subsys/bluetooth/mesh/rpr_srv.c

```diff
@@ -948,6 +948,10 @@ static int handle_link_close(struct bt_mesh_model *mod, struct bt_mesh_msg_ctx *
 		return -EINVAL;
 	}
 
+	if (reason != PROV_BEARER_LINK_STATUS_SUCCESS) {
+		atomic_clear_bit(bt_mesh_prov_link.flags, COMPLETE);
```

Comment:
I'm not sure that it is good to access the role's flags here. I'd rather recommend to throw this error to the role (`rpr_srv.c:handle_link_close() -> link_close() -> srv.refresh.cb->link_closed() -> prov.c:prov_link_closed() -> bt_mesh_prov_link.role->link_closed() -> prov_device.c:prov_link_closed()`) and process it there.